blivet/fstab.py

```diff
@@ -0,0 +1,792 @@
+        if self._table.intro_comment is not None:
+            clean_table.intro_comment = self._table.intro_comment
+        if self._table.trailing_comment is not None:
+            clean_table.trailing_comment = self._table.trailing_comment
+
+        entry = self._table.next_fs()
+        while entry:
+            if FSTabEntry(entry=entry).is_valid():
+                new_entry = self._copy_fs_entry(entry)
+                clean_table.add_fs(new_entry)
+            else:
+                log.warning("Fstab entry: '%s' is not complete, it will not be written into the file", entry)
+            entry = self._table.next_fs()
+
+        if os.path.exists(dest_file):
+            clean_table.replace_file(dest_file)
+        else:
+            try:
+                clean_table.write_file(dest_file)
+            except Exception as e:  # pylint: disable=broad-except
+                # libmount throws general Exception if underlying directories do not exist. Okay...
+                if str(e) == "No such file or directory":
+                    log.info("Underlying directory of fstab '%s' does not exist. creating...", dest_file)
+                    os.makedirs(os.path.split(dest_file)[0])
+                else:
+                    raise
+
+    def find_entry(self, spec=None, file=None, *, entry=None):
+        """ Return the line of loaded fstab with given spec and/or file.
+            If both entry and spec/file are given, spec/file are prioritized over entry values.
+
+            :keyword spec: searched device specs (see man fstab(5) fs_spec)
+            :type spec: str
+            :keyword file: device mount path (see man fstab(5) fs_file)
+            :type file: str
+            :keyword entry: fstab entry as an alternative input type
+            :type: :class: `FSTabEntry`
+            :returns: found fstab entry object
+            :rtype: :class: `FSTabEntry` or None
+        """
+
+        _spec = spec or (entry.spec if entry is not None else None)
+        _file = file or (entry.file if entry is not None else None)
+
+        found_entry = None
+
+        if _spec is not None and _file is not None:
+            try:
+                found_entry = self._table.find_pair(_spec, _file, MNT_ITER_FORWARD)
+            except LibmountException:
+                return None
+            return FSTabEntry(entry=found_entry)
+
+        if _spec is not None:
+            try:
+                found_entry = self._table.find_source(_spec, MNT_ITER_FORWARD)
+            except LibmountException:
+                return None
+            return FSTabEntry(entry=found_entry)
+
+        if file is not None:
+            try:
+                found_entry = self._table.find_target(file, MNT_ITER_FORWARD)
+            except LibmountException:
+                return None
+            return FSTabEntry(entry=found_entry)
+
+        return None
+
+    def get_spec(self, device):
```

Comment:
Do we need to have this as a public function? Looks like it is useful only internally.

Reply:
Fixed. Renamed to `_get_spec`.